src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -666,7 +666,18 @@ private static ulong TransferDataFromRequestStream(IntPtr buffer, int length, Ea
                     // Is there a previous read that may still have data to be consumed?
                     if (sts._task != null)
                     {
-                        Debug.Assert(sts._task.IsCompleted, "The task must have completed if we're getting called back.");
+                        if (!sts._task.IsCompleted)
+                        {
+                            // We have a previous read that's not yet completed.  This should be quite rare, but it can
+                            // happen when we're unpaused prematurely, potentially due to the request still finishing
+                            // being sent as the server starts to send a response.  Since we still have the outstanding
+                            // read, we simply re-pause.  When the task completes (which could have happened immediately
+                            // after the check). the continuation we previously created will fire and queue an unpause.
+                            // Since all of this processing is single-threaded on the current thread, that unpause request 
+                            // is guaranteed to happen after this re-pause.
+                            multi.VerboseTrace("Re-pausing reading after a spurious un-pause", easy: easy);
```

Comment:
Right now it's printf :smile: (when compiled with CURLHANDLER_VERBOSE enabled, otherwise it's a nop).  Eventually we'd want to standardized on whatever tracing mechanism is used by the rest of HttpClient, but printf is still the easiest way of getting such info on Linux/OSX today.